src/components/atoms/Timeline/index.tsx

```diff
@@ -0,0 +1,244 @@
+  renderKnob: () => ReactElement;
+  knobSize: number;
+};
+
+const EPOCH_SEC = 1000;
+const PADDING_HORIZONTAL = 12;
+const STRONG_MEMORY_HOURS = 3;
+const STRONG_MEMORY_WIDTH = 2;
+const NORMAL_MEMORY_WIDTH = 1;
+const GAP_HORIZONTAL = 14;
+const HOURS_SECS = 3600;
+const MEMORY_INTERVAL = 600;
+const DAY_SECS = 86400;
+
+const getRange = (_range: Props["range"]): Range => {
+  const { start, end } = _range || {};
+  if (start !== undefined && end !== undefined) {
+    return { start, end };
+  }
+
+  if (start !== undefined && end === undefined) {
+    return {
+      start,
+      end: start + DAY_SECS,
+    };
+  }
+
+  if (start === undefined && end !== undefined) {
+    return {
+      start: Math.max(end - DAY_SECS, 0),
+      end,
+    };
+  }
+
+  const defaultStart = Date.now();
+
+  return {
+    start: defaultStart,
+    end: defaultStart + DAY_SECS * EPOCH_SEC,
+  };
+};
+
+const Timeline: React.FC<Props> = memo(
+  function TimelinePresenter({ currentTime, range: _range, renderKnob, knobSize }) {
+    const range = useMemo(() => {
+      const range = getRange(_range);
+      if (process.env.NODE_ENV !== "production") {
+        if (range.start > range.end) {
+          throw new Error("Out of range error. `range.start` should be less than `range.end`");
+        }
+      }
+      return range;
+    }, [_range]);
+    const { start, end } = range;
+    const epochDiff = end - start;
+
+    // convert epoch diff to second.
+    const memoryCount = useMemo(
+      () => Math.trunc(epochDiff / EPOCH_SEC / MEMORY_INTERVAL),
+      [epochDiff],
+    );
+
+    const startDate = useMemo(() => new Date(start), [start]);
+    const hoursCount = Math.trunc(HOURS_SECS / MEMORY_INTERVAL);
+    // Convert memory count to pixel.
+    const currentPosition = useMemo(() => {
+      const diff = Math.min((currentTime - start) / EPOCH_SEC / MEMORY_INTERVAL, memoryCount);
+      const strongMemoryCount = diff / (hoursCount * STRONG_MEMORY_HOURS);
+      return Math.max(
+        diff * GAP_HORIZONTAL +
+          (diff - strongMemoryCount) * NORMAL_MEMORY_WIDTH +
+          strongMemoryCount * STRONG_MEMORY_WIDTH +
+          STRONG_MEMORY_WIDTH / 2,
+        0,
+      );
+    }, [currentTime, start, memoryCount, hoursCount]);
+
+    return (
+      <Container>
+        <MemoryBox>
+          <MemoryList start={startDate} memoryCount={memoryCount} hoursCount={hoursCount} />
+          <Icon
+            style={{
+              left: currentPosition + PADDING_HORIZONTAL - knobSize / 2,
+            }}>
+            {renderKnob()}
+          </Icon>
+        </MemoryBox>
+      </Container>
+    );
+  },
+  (prev, next) => prev.currentTime === next.currentTime,
+);
+
+const Container = styled.div`
+  background: ${({ theme }) => theme.main.lighterBg};
+  width: 100%;
+  display: flex;
+  align-items: center;
+  padding: 5px 12px;
+  box-sizing: border-box;
+`;
+
+const MemoryBox = styled.div`
+  border: 1px solid #b5b5b5;
```

Comment:
Please make sure you are always using the theme's colors and not hardcoding.

Reply:
Can I add `#b5b5b5` to theme? I could not find this color in theme.

Reply:
Sorry for the late reply...! I guess last review I did I jumped to the code and didn't check these replies...
Of course. If @lavalse (our designer) adds any colors to a design that you are to implement and they aren't already in the theme you can add them to the theme. 
Moreover, if you have time, or just notice that the color is very close to another color already in the system, you can also ask him to verify whether he wants the new color or stick to a color we are already using.

Reply:
This color is fixed because design is fixed:pray:
https://github.com/reearth/reearth-web/blob/bbf9ff24381e83888073834172cfa30f48180bef/src/components/atoms/Timeline/index.tsx#L179